src/presentation/http/router/note.test.ts

```diff
@@ -385,6 +385,65 @@ describe('Note API', () => {
     test.todo('Returns 400 when parentId has incorrect characters and lenght');
   });
 
+  describe('DELETE /note/:notePublicId', () => {
+    test('Should remove all note relations contains note id', async () => {
+      const accessToken = global.auth(2);
+      const correctID = 'f43NU75weU';
+
+      let response = await global.api?.fakeRequest({
+        method: 'DELETE',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      /**
+       * Checks if the note itself has been deleted
+       */
+      response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+
+      const childNote = 'Uyd8TgkdA0';
+
+      /**
+       * Checks if the note saved as a parent has been deleted
+       */
+      response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${childNote}`,
+      });
+
+      const expectedResponse = {
+        'note': {
+          'id': 56,
+          'publicId': 'Uyd8TgkdA0',
+          'creatorId': 2,
+          'content': null,
+          'createdAt': '2023-10-16T13:49:19.000Z',
+          'updatedAt': '2023-10-16T13:49:19.000Z',
+        },
+        'accessRights': {
+          'canEdit': true,
+        },
+      };
+
+      expect(response?.json()).toStrictEqual(expectedResponse);
```

Comment:
it's not clear what do you check here

Reply:
We don't have a direct request to fetch any note relationship - we do it via a GET request on the `/note/:notePublicId `route. If the note has a parent, it is returned with it. If it doesn't - the note comes without the additional **parentNote** field.

If the note id was stored in the **noteId** column of the **noteRelations** table, we get the `'note not found'` message when deleting it from the tables. If the note was not deleted - we would get back the note and its parent, which is attached to it in the database.

If the id of the note was stored in the **parentNoteId** column of the **noteRelations** table, then after its deletion only its child note will be returned on the GET request.

This is how I check if the note's relations have been deleted. There is no other way to retrieve the relations.

Reply:
i think you can use .not.hasProperty instead https://vitest.dev/api/expect#tohaveproperty